src/status_im/transport/message/v1/protocol.cljs

```diff
@@ -63,11 +63,17 @@
 
 (defn- prepare-recipients [public-keys db]
   (map (fn [public-key]
-         (select-keys (get-in db [:transport/chats public-key]) [:topic :sym-key-id]))
+         (if-let [{:keys [topic sym-key-id]} (get-in db [:transport/chats public-key])]
+           {:topic      topic
+            :sym-key-id sym-key-id}
+           {:topic   (transport.utils/get-topic constants/contact-discovery)
+            :pub-key public-key}))
        public-keys))
 
 (defn multi-send-with-pubkey
-  "Sends payload to multiple participants selected by `:public-keys` key."
+  "Sends payload to multiple participants selected by `:public-keys` key.
+  If there is already established symmetric key with the participant, uses that (for efficiency),
+  if not, uses asymetric encryption."
```

Comment:
since the admin only adds his contacts to the group there should always be an established symmetric key why is this change needed ?

Reply:
1. We are using this method to send contact updates as well (potentially to people who you have in contacts but you have no open chats with them, so no symmetric key)
2. `multi-send-with-pubkey` is used to actually send group key to all the participants (`NewGroupKey`), for example when new group chat is first established, or new person is added - we must use asymmetric encryption there

Reply:
Ok makes sense for contact updates, but for NewGroupKey I don't see why we can't send with symmetric encryption since the admin is a contact of all members. We should relabel the function multi-send since it's not only sending with pubkey anymore though

Reply:
I don't want to make assumption that when I'm somebody's contact, I have always chat with negotiated symmetric key opened with him - for example once I delete the chat, I don't, even when we are still contacts. 
I agree about renaming the function plain, `multi-send` is much less confusing.

Reply:
If you don't want to make that assumption then you should always send with public key ?

Reply:
That's probably cleaner code wise, but using the existing symmetric key established for (potentially) existing 1-1 chat with particular contact is just a minor perf optimisation.